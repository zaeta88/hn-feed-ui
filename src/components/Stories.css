h1 {
  font-size: 5rem;
}

.content {
  background-color: #fff;
  border-radius: 3px;
}

.right-button {
  margin-right: 15px;
  padding: 5px 10px;
}

.title-bar {
  background-color: gray;
  color: white;
  padding: 8px;
  border-radius: 0 0 3px 3px;
  height: 180px;
}

.react-confirm-alert-body {
  font-family: "Inconsolata", monosapce;
  color: #000;
}

.actions .member-link {
  border-radius: 3px;
  display: inline-block;
  cursor: pointer;
  padding: 0 0 5px 0;
  margin: 0;
  background-color: transparent;
  text-decoration: none;
}

.actions .member-link.disabled, .actions .member-link:disabled {
  opacity: 0.6;
  cursor: no-drop;
}
.actions .member-link:hover {
  background-color: #e1fce5;
}

.actions .member-link:disabled {
  background-color: gray;
}

.actions .member-link img {
  height: 22px;
}

.btn-secondary.dropdown-toggle, .react-confirm-alert-body > .react-confirm-alert-button-group button {
  color: #28a745;
  background-color: transparent;
  background-image: none;
  border: 1px solid #28a745;
  margin-right: 10px;
}

.btn-secondary:hover.dropdown-toggle, .btn-secondary:active.dropdown-toggle, .react-confirm-alert-body > .react-confirm-alert-button-group button:hover, .react-confirm-alert-body > .react-confirm-alert-button-group button:active {
  color: #fff;
  background-color: #28a745;
}


.pagination, .right-button, .actions, .carret {
  float: right;
}

.page-item.active .page-link {
  background-color: #5bc66a;
  border-color: #5bc66a;
}

.page-link {
  color: #5bc66a;
}

.search-label, .item {
  margin-top: 5px;
  width: 100%;
}


.table-bordered tbody tr:first-child td {
    padding-top: .7rem;
}

.table-bordered {
  border: none;
}

.table-bordered td, .table-bordered th {
  border: none;
  padding: .35rem .3rem .3rem .75rem;
  border-bottom: 1px solid #ccc;
}

td a.MuiLink-underlineHover {
  display: inline-block;
  width: 100%;
  color: #333;
}

td a.MuiLink-underlineHover:hover {
  text-decoration: none;
  color: #333;
}

.author {
  color: #999;
}

.table-bordered thead th {
  border-bottom: 2px solid #5bc66a;
}

.actions button.delete-btn {
  margin-right: 12px;
}

.content-wrapper {
  padding: 10px 24px;
}

.item img {
  height: 18px;
}