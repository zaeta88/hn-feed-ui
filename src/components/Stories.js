import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link, Button } from '@material-ui/core';
import BootstrapTable from 'react-bootstrap-table-next';
import ToolkitProvider from 'react-bootstrap-table2-toolkit';
import paginationFactory from 'react-bootstrap-table2-paginator';
import { confirmAlert } from 'react-confirm-alert';
import DeleteAlert from './DeleteAlert';
import trash from '../assets/img/trash.svg';
import './Stories.css';
import 'react-confirm-alert/src/react-confirm-alert.css';
import * as moment from 'moment';

class Stories extends Component {
  tableColumns = () => {
    const columns = [{
      dataField: 'title',
      text: 'Title',
      formatter: this.titleFormatter
    }, {
      dataField: 'date',
      text: 'Date',
      formatter: this.dateFormatter
    }, {
      dataField: '',
      text: '',
      align: 'center',
      formatter: this.deleteButtonFormatter,
      events: {
        onClick: (e, column, columnIndex, row, rowIndex) => {
          confirmAlert({
            customUI: ({ onClose }) => {
              return (
                <DeleteAlert 
                  handleDelete = { this.props.handleDelete }
                  onClose = { onClose }
                  row = { row }
                />
              );
            }
          });
        }
      }
    }];
    return columns;
  }

  tablePaginationOptions = (stories) => {
    const options = {
      paginationSize: 5,
      pageStartIndex: 1,
      showTotal: true,
      paginationTotalRenderer: this.customTotal,
      sizePerPageList: [{
        text: '5', value: 5
      }, {
        text: '10', value: 10
      }, {
        text: '25', value: 25
      }, {
        text: 'All Stories', value: stories.length
      }] 
    };
    return options;
  }
  
  dateFormatter = (cell, row) => {
    let date = moment(row.date).calendar(null, {
      sameDay: 'LT',
      nextDay: 'Tomorrow',
      nextWeek: 'dddd',
      lastDay: '[Yesterday]',
      lastWeek: 'MMM DD',
      sameElse: 'MMM DD'
    });
    return (
      <span className="">{date}</span>
    );
  }

  deleteButtonFormatter = (cell, row) => {
    return (
      <div className="actions mb-0">
        <Button className="member-link btn delete-btn" title='Delete'>
          <img src={trash} alt='Delete, trash can icon.'></img>
        </Button>
      </div>
    );
  }

  titleFormatter = (cell, row) => {
    return (
      <Link href={row.url}>
        <span className="title">{row.title}</span> - <span className="author">{row.author}</span> -
      </Link>
    );
  }
  
  render() {
    let stories = this.props.stories;

    return (
      <div className="stories">
        <div className="site-section pb-0">
          <div className="title-bar row">
            <div className="titlebar-left col-md-6">
              <h1>HN Feed</h1>
              <h2 className="title">{'We <3 Hacker News'}</h2>
            </div>
          </div>
          <div className="content row mt-4">
            <div className="content-wrapper container">
              <ToolkitProvider
                keyField="id"
                data={ stories }
                columns={ this.tableColumns() }
              >
                {
                  props => (
                    <div>
                      <BootstrapTable
                        { ...props.baseProps }
                        noDataIndication="No news to be showed."
                        pagination={ paginationFactory(this.tablePaginationOptions(stories)) }
                        hover
                      />
                    </div>
                  )
                }
              </ToolkitProvider>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Stories.propTypes = {
  stories: PropTypes.array
};

export default Stories;