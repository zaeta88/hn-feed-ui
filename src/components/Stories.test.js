import React from 'react';
import { shallow } from 'enzyme';
import Stories from './Stories';


describe('React.Component Stories', () => {
  it('should render correctly', () => {
    const component = shallow(<Stories />);
    expect(component).toMatchSnapshot();
  });
});
