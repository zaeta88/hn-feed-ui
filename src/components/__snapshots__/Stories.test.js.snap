// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React.Component Stories should render correctly 1`] = `ShallowWrapper {}`;
